lib/workers/repository/update/branch/index.ts

```diff
@@ -79,8 +80,34 @@ export interface ProcessBranchResult {
   result: BranchResult;
 }
 
+export function canSkipBranchUpdateCheck(
+  branchName: string,
+  branchCache: BranchCache,
+  branchFingerprint: string
+): boolean {
+  const branchCommitSha = getBranchCommit(branchName);
+
+  if (!branchCache.branchFingerprint) {
+    logger.debug('Branch fingerprint not found in cache, cannot skip branch');
```

Comment:
Let's not log anything at all here if caching is not enabled. Otherwise it's just noise

Reply:
We should distinguish between the cases of (a) no caching, and (b) caching but no fingerprint

Reply:
Should the no cache condition be caught and handled here or in file `write.ts` where we initialise `branchCache`?